### ISSUE TITLE
Container Include/exclude by tab "gmail.com" apply also on URL https://sub.domain.com/file.php?n=&rt=0&r=&r2=&st=0&em=name%40gmail.com&ip...

### ISSUE THREAD

**Issue by slrslr** [mbnuqw/sidebery#1447]
In Firefox-ESR I have Sidebery with a container SOCKS5 proxy set. And when i search at https://sub.domain.com/search.php
It switch to results with URL in address bar:

`https://sub.domain.com/search.php?n=&rt=0&r=&r2=&st=0&em=name%40gmail.com&ip=&as=0&co=&ma=&c=3&dt=0&d=&d2=&ult=0&ul=&ul2=&do=0&lst=0&ls=&ls2=&dlt=0&dl=&dl2=&w=0&submit=Submit+Query`

and it redirects to a new container tab thanks to a "Include/exclude by tab" rule:

`,{"id":"somestring","active":true,"type":1,"url":"gmail.com"},{"id":"somestring","active":true,"type":1,"url":"gmodules.com"},{"id":"somestring","active":true,"type":1,"url":"gogle.com"}`
(i think it is the first include type rule of the three)

I suspect it should not happen, one expect that the gmail.com is applied only to domains, not parameters of the URL.

I have tried to add search.php and sub.domain.com at the end of the list of incluides/excludes in aim to override some of the hundreds rules i have, but that not worked. Even exporting rules, editing file placing exclude rule "sub.domain.com/search.php" as the first one (i am unable to click to move lastly added rule to the top of the list (assuming first is the top priority when it not worked on the end of the list).
Yet i was still redirected to a container tab until removing mentioned gmail.com include rule.

Can something be improved about this? What is theworkaround to containerize gmail.com while preventing redirect on sub.domain.com/search.php?....gmail.com....     ?

WORKAROUND that seems to be working:
`/^https?:\/\/([^/]*\.)?gmail\.com\//`

**Comment by devgid** [comment]
Why don't you change the rule to be https://www.gmail.com or https://mail.google.com ?

**Comment by slrslr** [comment]
> Why don't you change the rule to be https://www.gmail.com or https://mail.google.com ?

Because i expected that it is more efficient (for example when it comes to space used by the text and line count) to use domain instead of two full addresses. actually i would expect that when i enter gmail.com, it will automatically cover:
```
http://gmail.com
https://gmail.com
http://www.gmail.com
https://www.gmail.com
```
but not:
`http(s)://sub.gmail.com`

so when i have these Sidebery rules:
```
*.gmail.com
https://gmail.com
```
and enter into a Firefox address bar: `gmail.com`, it does NOT open the address in container, because it seems to need Sidebery HTTP rule `http://gmail.com` rule. It should be sufficing to enter just `gmail.com` rule without need to bother with http/s and when i want to match in the middle of the URL, i would do *gmail.com*.
But i understand it is already made the way like it is.

So in short, i need to convert
```
*.domain.tld
domain.tld
```

to:
```
http://www.domain.tld
https://www.domain.tld
http://domain.tld
https://domain.tld
```

(doubling my list size) is that correct? Thank you

UPDATE: this seems to be working
`/^https?:\/\/([^/]*\.)?gmail\.com\//`

**Comment by devgid** [comment]
Then use regular expressions
Something like this:

Contains gmail.com but not if that string has he character "&" preceding it anywhere to the left of the string
>^(?!.*&).*gmail\.com$

or
>^(?=.*gmail\.com)(?<!&.*)$

You could add all the variations in one line that precedes it but that kinda gets complex
>^(https?:\\/\\/www\\.|https?:\\/\\/|http:\\/\\/www\\.|http:\\/\\/)?[a-zA-Z0-9.-]+\\.gmail\\.com$

**Comment by mbnuqw** [comment]
Expected behavior. Use regexp or "Configure site..." option in context menu.
For simplification of adding rules with matching by domain, track the #1384
